src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionItemFilter.cs

```diff
@@ -0,0 +1,102 @@
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Tags;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data;
+using Microsoft.VisualStudio.Text.Adornments;
+using RoslynCompletionItem = Microsoft.CodeAnalysis.Completion.CompletionItem;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+{
+    internal static class CompletionItemFilter
+    {
+        private static readonly ImmutableDictionary<string, CompletionFilter> allFilters;
+
+        static CompletionItemFilter()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, CompletionFilter>();
+
+            NamespaceFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Namespaces, WellKnownTags.Namespace, 'n', builder);
+            ClassFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Classes, WellKnownTags.Class, 'c', builder);
+            ModuleFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Modules, WellKnownTags.Module, 'u', builder);
+            StructureFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Structures, WellKnownTags.Structure, 's', builder);
+            InterfaceFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Interfaces, WellKnownTags.Interface, 'i', builder);
+            EnumFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Enums, WellKnownTags.Enum, 'e', builder);
+            DelegateFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Delegates, WellKnownTags.Delegate, 'd', builder);
+            ConstantFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Constants, WellKnownTags.Constant, 'o', builder);
+            FieldFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Fields, WellKnownTags.Field, 'f', builder);
+            EventFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Events, WellKnownTags.Event, 'v', builder);
+            PropertyFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Properties, WellKnownTags.Property, 'p', builder);
+            MethodFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Methods, WellKnownTags.Method, 'm', builder);
+            ExtensionMethodFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Extension_methods, WellKnownTags.ExtensionMethod, 'x', builder);
+            LocalAndParameterFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Locals_and_parameters, ImmutableArray.Create(WellKnownTags.Local, WellKnownTags.Parameter), 'l', builder);
+            KeywordFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Keywords, ImmutableArray.Create(WellKnownTags.Keyword), 'k', builder);
```

Comment:
nit: I think the last three calls can use another `CreateCompletionFilterAndAddToBuilder` overload